src/ClassicalSeekingMode.ts

```diff
@@ -0,0 +1,109 @@
+        if (cat.Position === undefined) {
+            console.error('catpos undefined');
+        }
+        const catPos = cat.Position;
+        if (this.countsOfDimensionsToChange == 1) {
+            const dimension = this.mersenneTwister.random();
+            const randomMove = (pos: Position, x, y) => ((this.mersenneTwister.random() > 0.5) ? pos.add : pos.subtract)(x, y);
+            if (dimension < 0.5) {
+                cat.Position = randomMove(catPos, catPos.x * this.seekingRangeOfSelectedDimension, 0);
+            } else {
+                cat.Position = randomMove(catPos, 0, catPos.y * this.seekingRangeOfSelectedDimension);
+            }
+        }
+        if (this.countsOfDimensionsToChange == 2) {
+            const randomOffset = (value, offset) => value + (this.mersenneTwister.random() < 0.5) ? -offset : offset;
+            cat.Position = new Position(
+                randomOffset(catPos.x, catPos.x * this.seekingRangeOfSelectedDimension),
+                randomOffset(catPos.y, catPos.y * this.seekingRangeOfSelectedDimension)
+            );
+        }
+    }
+
+    private calculateSelectionProb(copies: Cat[], allTheSame: Boolean, fitnessMax: number, fitnessMin: number): void {
+        if (allTheSame) {
+            copies.forEach(cat => cat.SelectionProb = 1);
+        } else {
+            copies.forEach(cat =>
+                cat.SelectionProb = Math.abs((cat.calculateFitness() - fitnessMin) / (fitnessMax - fitnessMin))
+            );
+        }
+    }
+
+    private chooseNewPosition(copies: Cat[]): Position {
+        let selectedCat: number;
+        let probability = []
+        probability[0] = 0;
+        for (let i = 0; i < this.seekingMemoryPool; i++) {
+            probability[i + 1] = copies[i].SelectionProb + probability[i];
+        }
+        let probhigh = probability[this.seekingMemoryPool - 1];
+        for (let i = 0; i < this.seekingMemoryPool; i++) {
+            probability[i] = probability[i] / probhigh;
+        }
+        const twist = this.mersenneTwister.random();
+        for (let i = 0; i < probability.length; i++) {
+            if (probability[i] > twist) {
+                selectedCat = i - 1;
+                break;
+             }
+       }
+        return copies[selectedCat].Position;
+    }
+
+    seek(cat: Cat, fitnessMax: number, fitnessMin: number): void {
+        let allTheSame: Boolean;
+        this.fitnessValues = new Array(this.seekingMemoryPool);
+        const copies: Cat[] = this.createCopies(cat);
+        for (let i = 0; i < this.j; i++) {
+            this.changePosition(copies[i]);
+            this.fitnessValues[i] = copies[i].calculateFitness();
+        }
+        if (this.selfPositionConsidering) {
+            this.fitnessValues[this.j] = copies[this.j].calculateFitness();
+        }
+        for (let i = 1; i < this.fitnessValues.length; i++) {
+            allTheSame = (this.fitnessValues[i - 1] === this.fitnessValues[i]);

```

Comment:
variable allTheSame really necessary?

Reply:
put check into the method, so the variable doesn't need to be tranferred